ptzclient/client.go

```diff
@@ -0,0 +1,517 @@
+)
+
+const (
+	defaultPanSpeed  = 0.5
+	defaultTiltSpeed = 0.5
+	defaultZoomSpeed = 0.5
+)
+
+// Model is the model for the ONVIF PTZ client.
+var Model = viamrtsp.Family.WithModel("onvif-ptz-client")
+
+func init() {
+	resource.RegisterComponent(
+		generic.API,
+		Model,
+		resource.Registration[resource.Resource, *Config]{
+			Constructor: newOnvifPtzClientClient,
+		},
+	)
+}
+
+// Config represents the configuration for the ONVIF PTZ client.
+type Config struct {
+	Address      string `json:"address"`
+	Username     string `json:"username"`
+	Password     string `json:"password"`
+	ProfileToken string `json:"profile_token"`
+}
+
+// Validate validates the configuration for the ONVIF PTZ client.
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.Address == "" {
+		return nil, fmt.Errorf(`expected "address" attribute for %s %q`, Model.String(), path)
+	}
+
+	if cfg.Username == "" {
+		return nil, fmt.Errorf(`expected "username" attribute for %s %q`, Model.String(), path)
+	}
+	if cfg.Password == "" {
+		return nil, fmt.Errorf(`expected "password" attribute for %s %q`, Model.String(), path)
+	}
+	return nil, nil
+}
+
+type onvifPtzClient struct {
+	resource.AlwaysRebuild
+
+	name   resource.Name
+	logger logging.Logger
+	cfg    *Config
+	dev    *onvif.Device
+
+	cancelCtx  context.Context
+	cancelFunc func()
+}
+
+func newOnvifPtzClientClient(
+	ctx context.Context,
+	deps resource.Dependencies,
+	rawConf resource.Config,
+	logger logging.Logger,
+) (resource.Resource, error) {
+	conf, err := resource.NativeConfig[*Config](rawConf)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewClient(ctx, deps, rawConf.ResourceName(), conf, logger)
+}
+
+// NewClient creates a new ONVIF PTZ client.
+func NewClient(
+	_ context.Context,
+	_ resource.Dependencies,
+	name resource.Name,
+	conf *Config,
+	logger logging.Logger,
+) (resource.Resource, error) {
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+
+	logger.Debugf("Attempting to connect to ONVIF device at %s", conf.Address)
+	dev, err := onvif.NewDevice(onvif.DeviceParams{
+		Xaddr:    conf.Address,
+		Username: conf.Username,
+		Password: conf.Password,
+	})
+	if err != nil {
+		cancelFunc()
+		return nil, fmt.Errorf("failed to create ONVIF device for %s: %w", conf.Address, err)
+	}
+	logger.Info("successfully connected to ONVIF device.")
+
+	s := &onvifPtzClient{
+		name:       name,
+		logger:     logger,
+		cfg:        conf,
+		dev:        dev,
+		cancelCtx:  cancelCtx,
+		cancelFunc: cancelFunc,
+	}
+	if s.cfg.ProfileToken == "" {
+		logger.Warn("No 'profile_token' configured. PTZ commands may fail. Run 'get-profiles' to discover available profiles.")
+	}
+
+	return s, nil
+}
+
+func (s *onvifPtzClient) Name() resource.Name {
+	return s.name
+}
+
+// handleGetProfiles retrieves available media profiles from the camera and implements the get-profiles command logic.
+func (s *onvifPtzClient) handleGetProfiles() (map[string]interface{}, error) {
```

Comment:
Could we just re-use GetProfiles from device.go?

Reply:
I think that would be nice but the `Device` in this instance is from my fork, not the one in `viamonvif`